ui/store/actions.ts

```diff
@@ -3351,6 +3351,21 @@ export function setUseNonceField(
   };
 }
 
+export function setPrivacyMode(
+  val: boolean,
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
+  return (dispatch: MetaMaskReduxDispatch) => {
+    dispatch(showLoadingIndication());
```

Comment:
Do we need to have a loading indicator here? Did you see any visual bugs that required this? I wonder if it would be smoother UX without it.

Reply:
Yeah, I'd avoid this tbh; it will likely always flash very quickly and be a gross UI flicker.  Let's remove it.

Reply:
In the [setPreference](https://github.com/MetaMask/metamask-extension/blob/ad6d1b3dde9e507df953bb65d88773a4519afd5c/ui/store/actions.ts#L3033) action, we can pass an optional boolean to indicate whether or not it should render the loading indicator, it might make sense to just use that one instead?